File: kamacoder44/kamacoder44.cpp
#include<iostream>
#include<cmath>
using namespace std;
const int N = 110;
int s[N][N];
int n, m;
int ans = 1e9;
void solve()
{
    for (int i = 1; i <= n; i++)
        for (int j = 1; j <= m; j++)
        {
            int x;
            cin >> x;
            s[i][j] = s[i - 1][j] + s[i][j - 1] - s[i - 1][j - 1] + x;
        }
    
    for (int i = 1; i < n; i++){
        int x = abs(s[n][m] - 2 * s[i][m]);
        ans = min(ans, x);
    }
    for (int j = 1; j < m; j++){
        int x = abs(s[n][m] - 2 * s[n][j]);
        ans = min(ans, x);
    }
    cout << ans;
}
int main(){
    cin >> n >> m;
    solve();
    return 0;
}

File: kamacoder44/kamacoder44.py
data = input().strip().split()
n, m = int(data[0]), int(data[1])

matrix = [[None] * m for _ in range(n)]
for i in range(n):
    row = input().split()
    for j in range(m):
        matrix[i][j] = int(row[j])

N = 100
s = [[0] * (N + 10) for _ in range(N + 10)]
for i in range(1, n + 1):
    for j in range(1, m + 1):
        s[i][j] = s[i - 1][j] + s[i][j - 1] - s[i - 1][j - 1] +  matrix[i - 1][j - 1]


ans = float('inf')

for i in range(1, n):
    x = abs(s[n][m] - 2 * s[i][m])
    ans = min(ans, x)
    
for j in range(1, m):
    x = abs(s[n][m] - 2 * s[n][j])
    ans = min(ans, x)

print(ans)